src/gwt/src/org/rstudio/core/client/VirtualConsole.java

```diff
@@ -75,18 +82,63 @@ public boolean screenReaderEnabled()
       {
          return getUserPrefs().enableScreenReader().getValue();
       }
+
+      @Override
+      public boolean limitConsoleVisible()
+      {
+         return getUserPrefs().limitVisibleConsole().getValue();
+      }
    }
 
    @Inject
    public VirtualConsole(@Assisted Element parent, final Preferences prefs)
    {
       prefs_ = prefs;
       parent_ = parent;
+
+      // only load the virtual scroller if enabled in prefs
+      if (prefs_.limitConsoleVisible() &&
+          parent_ != null &&
+          !VirtualConsole.virtualScrollerInitialized_ &&
```

Comment:
Nit: `VirtualConsole.` prefix is unnecessary (here and elsewhere)

Reply:
I really like prefixing it to make it explicit that we're referencing something static but if it's not how we do it in our style guide that's fine.